providers/dns/azure/azure.go

```diff
@@ -242,6 +244,10 @@ func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
 
 // Checks that azure has a zone for this domain name.
 func (d *DNSProvider) getHostedZoneID(ctx context.Context, fqdn string) (string, error) {
+	if env.GetOrFile(EnvZoneName) != "" {
+	    return env.GetOrFile(EnvZoneName), nil
```

Comment:
Since `GetOrFile` might read a file, its better to cache that return value:

```suggestion
	if zone := env.GetOrFile(EnvZoneName); zone != "" {
	    return zone, nil
```

Performance doesn't matter here, but this is more idiomatic Go code.